contracts/token/CRC1155/extensions/CRC1155Enumerable.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./ICRC1155Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+abstract contract CRC1155Enumerable is ERC1155, ICRC1155Enumerable {
+    using EnumerableSet for EnumerableSet.UintSet;
+
+    // All token ids for enumeration.
+    EnumerableSet.UintSet private _allTokens;
+
+    // Mapping from token ID to token total supply.
+    mapping(uint256 => uint256) private _totalSupplies;
+
+    // Mapping from owner to owned token IDs for enumeration.
+    mapping(address => EnumerableSet.UintSet) private _ownedTokens;
+
+    /**
+     * @dev Returns the total amount of tokens stored by the contract.
+     */
+    function totalSupply() public view virtual override returns (uint256) {
```

Comment:
I think this naming (and also *"amount of tokens"*) will lead to confusion.

Reply:
At least the doc can be changed to *"Returns the number of different tokenIds stored by the contract."*.

Reply:
Your suggestion is very good, I changed the comment as you mentioned above.